### PR TITLE
Add Mish with SoftPlus transformation

### PR DESCRIPTION
**Pull request by iimironov**

### CODE REVIEW

File: inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+ngraph::pass::SoftPlusToMishFusion::SoftPlusToMishFusion() {
+    auto input = ngraph::pattern::any_input();
+    auto softplus = std::make_shared<ngraph::opset4::SoftPlus>(input);
+    auto tanh = std::make_shared<ngraph::opset4::Tanh>(softplus);
+    auto mul = std::make_shared<ngraph::opset4::Multiply>(input, tanh);
+
+    ngraph::graph_rewrite_callback matcher_pass_callback = [=](ngraph::pattern::Matcher& m) {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp:20]:
```suggestion
    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
```

File: inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+#include "transformations/softplus_to_mish_fusion.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+ngraph::pass::SoftPlusToMishFusion::SoftPlusToMishFusion() {
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp:14]: Let's create GraphRewrite that will register all types of Mish fusions.

File: inference-engine/src/transformations/include/transformations/softplus_to_mish_fusion.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+namespace pass {
+
+class TRANSFORMATIONS_API SoftPlusToMishFusion;
+
+}  // namespace pass
+}  // namespace ngraph
+
+/**
+ * @ingroup ie_transformation_common_api
+ * @brief MishFusion transformation replaces group of
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/include/transformations/softplus_to_mish_fusion.hpp:26]:
```suggestion
 * @brief SoftPlusToMishFusion transformation replaces group of
```

File: inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp (hunk 2 above)
Comment (GlebKazantaev) [reply inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp:14]: We can do it later.

**Comment by GlebKazantaev** [comment]
@iimironov why this transformation is not registered in CommonOptimizations?

**Comment by iimironov** [comment]
> @iimironov why this transformation is not registered in CommonOptimizations?
Fixed

File: inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    auto input = ngraph::pattern::any_input();
+    auto softplus = std::make_shared<ngraph::opset4::SoftPlus>(input);
```
Comment (GlebKazantaev) [inline inference-engine/src/transformations/src/transformations/softplus_to_mish_fusion.cpp:16]: Softplus and Tanh could have more than one consumer and this replacement can break something. So I suggest to use wrap_type and use predicate that checks consumers count

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

**Review by GlebKazantaev — APPROVED** [review]

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**